src/main/java/duke/parser/Parser.java

```diff
@@ -0,0 +1,151 @@
+package duke.parser;
+
+import duke.datevalidator.DateValidator;
+import duke.taskclass.DeadlineTask;
+import duke.taskclass.EventTask;
+import duke.taskclass.Task;
+import duke.ui.ConsoleUI;
+
+import java.text.ParseException;
+import java.text.SimpleDateFormat;
+import java.util.Date;
+
+public class Parser {
+    ConsoleUI ui;
+    boolean isBye;
+    DateValidator validator;
+
+    /**
+     * Constructor of the Parser.
+     * @param ui Handler for user interface.
+     * @param validator Valid date format for tasks.
+     */
+    public Parser(ConsoleUI ui, DateValidator validator) {
+        this.ui = ui;
+        this.isBye = false;
+        this.validator = validator;
+    }
+
+    /**
+     * Exception class for missing todoTask descriptor.
+     */
+    static class MissingTodoDescriptorException extends Exception {
+        public MissingTodoDescriptorException(String errorMessage) {
+            super(errorMessage);
+        }
+    }
+
+    /**
+     * Exception class for unknown input parameters.
+     */
+    static class UnknownInputParamException extends Exception {
+        public UnknownInputParamException(String errorMessage) {
+            super(errorMessage);
+        }
+    }
+
+    public boolean getIsBye() {
+        return this.isBye;
+    }
+
+    /**
+     * Parses lines of user input and outputs corresponding command.
+     * @param tasks List of tasks from file.
+     * @param taskIterator Integer to count number of tasks at a time.
+     * @return New number of tasks after any addition or deletion.
+     */
+    public Tuple2<Integer, String> parseInput(String input, Task[] tasks, int taskIterator) {
```

Comment:
The parseInput method here is a bit long. Maybe its better to separate some of the parts like the deadline case into its own handleDeadline method for readability?